### PR TITLE
Refactor deploy to mavencentral

### PR DESCRIPTION
**Pull request by willynganga**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+      - name: Setup Java
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'zulu'
+          java-package: 'jdk+fx'
+
+      - name: Sonar analysis
+        uses: gradle/gradle-build-action@b5126f31dbc19dd434c3269bf8c28c315e121da2
```
Comment (Thilo-Acrolinx) [inline .github/workflows/build.yml:28]: should be "842c587ad8aa4c68eeba24c396e15af4c2e9f30a"

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        with:
+          java-version: '8'
+          distribution: 'zulu'
+          java-package: 'jdk+fx'
+
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@56b90f209b02bf6d1deae490e9ef18b21a389cd4
+
+      - name: Gradle build
+        uses: gradle/gradle-build-action@b5126f31dbc19dd434c3269bf8c28c315e121da2
```
Comment (Thilo-Acrolinx) [inline .github/workflows/build.yml:54]: should be "842c587ad8aa4c68eeba24c396e15af4c2e9f30a"

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        with:
+          arguments: build
```
Comment (Thilo-Acrolinx) [inline .github/workflows/build.yml:56]: please add a line break at the end of the file

File: .github/workflows/release-snapshot.yml [hunk 4]
```diff
@@ -1,45 +1,18 @@
...
-
-      - name: Sonar analysis
-        uses: gradle/gradle-build-action@842c587ad8aa4c68eeba24c396e15af4c2e9f30a
-        env:
-          SONAR_TOKEN: ${{ secrets.SONARCLOUD_TOKEN }}
-        with:
-          arguments: jacocoTestReport sonar
-
-  Build:
+  Deploy:
```
Comment (Thilo-Acrolinx) [inline .github/workflows/release-snapshot.yml:6]: should be "Release Snapshot" I guess?

File: .github/workflows/release-snapshot.yml [hunk 5]
```diff
@@ -70,7 +40,7 @@ jobs:
       - name: Deploy artifacts
         id: deploy
         if: github.ref == 'refs/heads/main'
-        run: ./deploy.sh
+        run: ./gradlew release -Prelease.useAutomaticVersion=true --stacktrace --warning-mode all
```
Comment (Thilo-Acrolinx) [inline .github/workflows/release-snapshot.yml:43]: we have "org.gradle.warning.mode=all" in the gradle.properties file, do we really need this "--warning-mode all" here?

File: .github/workflows/release-snapshot.yml [hunk 6]
```diff
@@ -70,7 +40,7 @@ jobs:
       - name: Deploy artifacts
         id: deploy
         if: github.ref == 'refs/heads/main'
```
Comment (Thilo-Acrolinx) [inline .github/workflows/release-snapshot.yml:42]: what if we want to release from another branch?

File: .github/workflows/release-snapshot.yml [hunk 7]
```diff
@@ -83,4 +53,4 @@ jobs:
         if: steps.deploy.outputs.RELEASE == 'true' && github.ref == 'refs/heads/main'
         with:
           branch: gh-pages
-          folder: "docs/"
+          folder: "docs/"
```
Comment (Thilo-Acrolinx) [inline .github/workflows/release-snapshot.yml:56]: please add line break at the end of the file

File: .github/workflows/release.yml [hunk 8]
```diff
@@ -0,0 +1,56 @@
+name: Release Stable Version
+on:
+  workflow_dispatch:
+
+jobs:
+  Deploy:
```
Comment (Thilo-Acrolinx) [inline .github/workflows/release.yml:6]: should be "Release Stable"?

File: build.gradle [hunk 9]
```diff
@@ -77,22 +76,8 @@ def buildVersion = buildNumber != null ? buildNumber : 123
...
-	proc.waitFor()
-	if (branch.startsWith('main')) {
-		isMasterBranch = true
-	} else {
-		isMasterBranch = false
-	}
-}
 
-task createVersionPropertiesFile {
+tasks.register('createVersionPropertiesFile') {
```
Comment (Thilo-Acrolinx) [inline build.gradle:80]: why do we need this gradle task and this "versionJavaSDK.properties" file at all? We do have a property called "currentVersion" in the gradle.properties file, so isn't this redundant?

File: build.gradle [hunk 10]
```diff
@@ -185,89 +170,65 @@ def devEmail = "opensource@acrolinx.com"
...
 
-nexusPublishing {
-	connectTimeout = Duration.ofMinutes(5);
-	clientTimeout = Duration.ofMinutes(5);
-	repositories {
-		sonatype {
-			username = sonatypeUsername
-			password = sonatypePassword
-		}
+	svn {
```
Comment (Thilo-Acrolinx) [inline build.gradle:221]: svn? I thought we use git?

**Review by Thilo-Acrolinx — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 9 above)
Comment (willynganga) [reply build.gradle:80]: This version is displayed in the Sidebar

File: .github/workflows/release.yml
Comment (Thilo-Acrolinx) [inline .github/workflows/release.yml:1]: Maybe rename this file to "release-stable.yml" for consistency?

File: .github/workflows/release-snapshot.yml [hunk 11]
```diff
@@ -70,7 +40,7 @@ jobs:
       - name: Deploy artifacts
         id: deploy
         if: github.ref == 'refs/heads/main'
-        run: ./deploy.sh
+        run: ./gradlew release -Prelease.useAutomaticVersion=true --stacktrace
 
       - name: Bump version and push tag
```
[the remark this answers is not in the document]
Comment (willynganga) [reply .github/workflows/release-snapshot.yml:45]: Sure, I'll make the change.

**Pull request closed without merging**